xcat-inventory/xcclient/allien/resources/__init__.py

```diff
@@ -5,14 +5,30 @@
 # -*- coding: utf-8 -*-
 
 
-from flask import Blueprint
+from flask import Blueprint, request, abort
 from flask_restplus import Resource, Api
+from functools import wraps
+from ..invmanager import check_user_token
 
 api_bp = Blueprint('api', __name__, url_prefix='/api')
 api = Api(api_bp, version='2.0', title='xCAT API v2', prefix="/v2",
           description='RESTful API of xCAT',
 )
 
+def auth_request(function):
+    @wraps(function)
+    def check_token(*args, **kwargs):
+        try:
+            auth_header = request.headers.get('Authorization')
+            auth_token = auth_header.split(" ")[1]
+        except Exception:
+            abort(401)
+        #return make_response("Unauthorized requrest, please login first", 401)
+        flag = check_user_token(auth_token)
+        if not flag == 0:
+            abort(401)
+        return function(*args, **kwargs)
```

Comment:
We may put the user name and user type into `g` object, this is available per requesting.
In that, other method could use this as a globla variable by import `g`

Reply:
g is a global variable which can be used anywhere for a reqest: http://flask.pocoo.org/docs/1.0/api/#flask.g
This is a good place to store resources during a request.